### ISSUE TITLE
JdkPlugin - classReload() exception This web container has not yet been started

### ISSUE THREAD

**Issue by FiruzzZ** [HotswapProjects/HotswapAgent#250]
I getting this exception when I add a static method from another class (despite of.. the modifications are applied, everything seems to work fine)

example: 
```
public class MyView implements View {
...
new Button("Press here " + Project.getSomeString()); //where getSomeString is applied and works so far
...
}
public class Project {
public static String getSomeString() {
return "..............";
}
}
```

```
Información:   HOTSWAP AGENT: 08:19:02.288 RELOAD (org.hotswap.agent.plugin.jvm.AnonymousClassPatchPlugin) - Class 'com/traslado/controller/AsignacionTrasladoABMController' has been enhanced with anonymous classes for hotswap.
Información:   HOTSWAP AGENT: 08:19:08.906 RELOAD (org.hotswap.agent.plugin.jvm.AnonymousClassPatchPlugin) - Class 'com/traslado/controller/AsignacionTrasladoABMController' has been enhanced with anonymous classes for hotswap.
Información:   HOTSWAP AGENT: 08:19:08.975 RELOAD (org.hotswap.agent.plugin.jvm.AnonymousClassPatchPlugin) - Class 'com/traslado/controller/TrasladoController' has been enhanced with anonymous classes for hotswap.
Información:   HOTSWAP AGENT: 08:19:09.051 RELOAD (org.hotswap.agent.plugin.jvm.AnonymousClassPatchPlugin) - Class 'com/traslado/ui/TrasladoUI' has been enhanced with anonymous classes for hotswap.
Información:   HOTSWAP AGENT: 08:19:09.096 RELOAD (org.hotswap.agent.plugin.jvm.AnonymousClassPatchPlugin) - Class 'com/traslado/ui/LoginView' has been enhanced with anonymous classes for hotswap.
Información:   HOTSWAP AGENT: 08:19:09.117 ERROR (org.hotswap.agent.plugin.jdk.JdkPlugin) - classReload() exception This web container has not yet been started.
Información:   HOTSWAP AGENT: 08:19:09.117 ERROR (org.hotswap.agent.plugin.jdk.JdkPlugin) - classReload() exception This web container has not yet been started.
Información:   HOTSWAP AGENT: 08:19:09.127 ERROR (org.hotswap.agent.annotation.handler.PluginClassFileTransformer) - InvocationTargetException in transform method on plugin 'class org.hotswap.agent.plugin.proxy.ProxyPlugin' class 'com/traslado/ui/TrasladoUI$TrasladoUIServlet'.
java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor106.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.hotswap.agent.annotation.handler.PluginClassFileTransformer.transform(PluginClassFileTransformer.java:193)
	at org.hotswap.agent.annotation.handler.PluginClassFileTransformer.transform(PluginClassFileTransformer.java:89)
	at org.hotswap.agent.util.HotswapTransformer.transform(HotswapTransformer.java:184)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
Caused by: java.lang.IllegalStateException: This web container has not yet been started
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1544)
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1501)
	at org.hotswap.agent.plugin.proxy.hscglib.GeneratorParametersTransformer.getGeneratorParamsMap(GeneratorParametersTransformer.java:83)
	at org.hotswap.agent.plugin.proxy.hscglib.GeneratorParametersTransformer.getGeneratorParams(GeneratorParametersTransformer.java:110)
	at org.hotswap.agent.plugin.proxy.ProxyPlugin.transformCglibProxy(ProxyPlugin.java:111)
	... 8 more

Información:   HOTSWAP AGENT: 08:19:09.127 ERROR (org.hotswap.agent.annotation.handler.PluginClassFileTransformer) - InvocationTargetException in transform method on plugin 'class org.hotswap.agent.plugin.weld.WeldPlugin' class 'com/traslado/ui/TrasladoUI$TrasladoUIServlet'.
java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor107.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.hotswap.agent.annotation.handler.PluginClassFileTransformer.transform(PluginClassFileTransformer.java:193)
	at org.hotswap.agent.annotation.handler.PluginClassFileTransformer.transform(PluginClassFileTransformer.java:89)
	at org.hotswap.agent.util.HotswapTransformer.transform(HotswapTransformer.java:184)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
Caused by: java.lang.IllegalStateException: This web container has not yet been started
	at org.glassfish.web.loader.WebappClassLoader.findResourceInternal(WebappClassLoader.java:2739)
	at org.glassfish.web.loader.WebappClassLoader.findResource(WebappClassLoader.java:1188)
	at org.glassfish.web.loader.WebappClassLoader.getResource(WebappClassLoader.java:1305)
	at org.hotswap.agent.javassist.LoaderClassPath.find(LoaderClassPath.java:93)
	at org.hotswap.agent.javassist.ClassPoolTail.find(ClassPoolTail.java:361)
	at org.hotswap.agent.javassist.ClassPool.find(ClassPool.java:584)
	at org.hotswap.agent.javassist.ClassPool.createCtClass(ClassPool.java:568)
	at org.hotswap.agent.javassist.ClassPool.get0(ClassPool.java:533)
	at org.hotswap.agent.javassist.ClassPool.get(ClassPool.java:442)
	at org.hotswap.agent.javassist.CtClassType.getDeclaringClass(CtClassType.java:835)
	at org.hotswap.agent.plugin.weld.WeldPlugin.isInnerNonPublicStaticClass(WeldPlugin.java:251)
	at org.hotswap.agent.plugin.weld.WeldPlugin.classReload(WeldPlugin.java:197)
	... 8 more

Información:   HOTSWAP AGENT: 08:19:09.139 ERROR (org.hotswap.agent.plugin.jdk.JdkPlugin) - classReload() exception This web container has not yet been started.
Información:   HOTSWAP AGENT: 08:19:09.140 ERROR (org.hotswap.agent.plugin.jdk.JdkPlugin) - classReload() exception This web container has not yet been started.
Información:   HOTSWAP AGENT: 08:19:09.141 ERROR (org.hotswap.agent.annotation.handler.PluginClassFileTransformer) - InvocationTargetException in transform method on plugin 'class org.hotswap.agent.plugin.proxy.ProxyPlugin' class 'com/traslado/ui/TrasladoUI$TrasladoUIServlet'.
java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor106.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.hotswap.agent.annotation.handler.PluginClassFileTransformer.transform(PluginClassFileTransformer.java:193)
	at org.hotswap.agent.annotation.handler.PluginClassFileTransformer.transform(PluginClassFileTransformer.java:89)
	at org.hotswap.agent.util.HotswapTransformer.transform(HotswapTransformer.java:184)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
Caused by: java.lang.IllegalStateException: This web container has not yet been started
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1544)
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1501)
	at org.hotswap.agent.plugin.proxy.hscglib.GeneratorParametersTransformer.getGeneratorParamsMap(GeneratorParametersTransformer.java:83)
	at org.hotswap.agent.plugin.proxy.hscglib.GeneratorParametersTransformer.getGeneratorParams(GeneratorParametersTransformer.java:110)
	at org.hotswap.agent.plugin.proxy.ProxyPlugin.transformCglibProxy(ProxyPlugin.java:111)
	... 8 more

Información:   HOTSWAP AGENT: 08:19:09.157 ERROR (org.hotswap.agent.annotation.handler.PluginClassFileTransformer) - InvocationTargetException in transform method on plugin 'class org.hotswap.agent.plugin.weld.WeldPlugin' class 'com/traslado/ui/TrasladoUI$TrasladoUIServlet'.
java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor107.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.hotswap.agent.annotation.handler.PluginClassFileTransformer.transform(PluginClassFileTransformer.java:193)
	at org.hotswap.agent.annotation.handler.PluginClassFileTransformer.transform(PluginClassFileTransformer.java:89)
	at org.hotswap.agent.util.HotswapTransformer.transform(HotswapTransformer.java:184)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
Caused by: java.lang.IllegalStateException: This web container has not yet been started
	at org.glassfish.web.loader.WebappClassLoader.findResourceInternal(WebappClassLoader.java:2739)
	at org.glassfish.web.loader.WebappClassLoader.findResource(WebappClassLoader.java:1188)
	at org.glassfish.web.loader.WebappClassLoader.getResource(WebappClassLoader.java:1305)
	at org.hotswap.agent.javassist.LoaderClassPath.find(LoaderClassPath.java:93)
	at org.hotswap.agent.javassist.ClassPoolTail.find(ClassPoolTail.java:361)
	at org.hotswap.agent.javassist.ClassPool.find(ClassPool.java:584)
	at org.hotswap.agent.javassist.ClassPool.createCtClass(ClassPool.java:568)
	at org.hotswap.agent.javassist.ClassPool.get0(ClassPool.java:533)
	at org.hotswap.agent.javassist.ClassPool.get(ClassPool.java:442)
	at org.hotswap.agent.javassist.CtClassType.getDeclaringClass(CtClassType.java:835)
	at org.hotswap.agent.plugin.weld.WeldPlugin.isInnerNonPublicStaticClass(WeldPlugin.java:251)
	at org.hotswap.agent.plugin.weld.WeldPlugin.classReload(WeldPlugin.java:197)
	... 8 more

Información:   HOTSWAP AGENT: 08:19:09.171 RELOAD (org.hotswap.agent.plugin.jvm.AnonymousClassPatchPlugin) - Class 'com/traslado/controller/UsuarioController' has been enhanced with anonymous classes for hotswap.
Información:   HOTSWAP AGENT: 08:19:09.533 ERROR (org.hotswap.agent.plugin.elresolver.PurgeBeanELResolverCacheCommand) - Error purging BeanELResolver cache.
java.lang.IllegalStateException: This web container has not yet been started
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1544)
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1501)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at org.hotswap.agent.plugin.elresolver.PurgeBeanELResolverCacheCommand.executeCommand(PurgeBeanELResolverCacheCommand.java:31)
	at org.hotswap.agent.command.impl.CommandExecutor.run(CommandExecutor.java:25)

Información:   HOTSWAP AGENT: 08:19:09.533 ERROR (org.hotswap.agent.plugin.elresolver.PurgeBeanELResolverCacheCommand) - Error purging BeanELResolver cache.
java.lang.IllegalStateException: This web container has not yet been started
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1544)
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1501)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at org.hotswap.agent.plugin.elresolver.PurgeBeanELResolverCacheCommand.executeCommand(PurgeBeanELResolverCacheCommand.java:31)
	at org.hotswap.agent.command.impl.CommandExecutor.run(CommandExecutor.java:25)
```

**Comment by shoikot** [comment]
@FiruzzZ Or anyone  have you revolved this issue yet ?  I facing this too..

**Comment by skybber** [comment]
I have similar problem in Tomcat {glassfish has extended WebAppClassloader from Tomcat|, I have already [fixed it](https://github.com/HotswapProjects/HotswapAgent/commit/30984d378626f1b2ed2eef4739b158a7fb132598), fixes are in [v1.3.3-SNAPSHOT](https://github.com/HotswapProjects/HotswapAgent/releases/tag/1.3.1-SNAPSHOT). Are you using 1.3.1?

**Comment by shoikot** [comment]
thanks for the response, I'm using `v1.3.0` , I'm going to use the updated version from now on :)
